src/routes/blog/post/cors-error/+page.markdoc

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title: Solving CORS errors with Appwrite
+description: Understanding why you are getting CORS error when sending a request to an appwrite backend and how to debug.
+date: 2024-01-16
+cover: /images/blog/cors-error/cors_cover.png
+timeToRead: 3
+author: dennis-ivy
+category: engineering
+featured: false
+---
+
+I want to address an issue I've seen popping up on stack overflow and the appwrite discord server and address some of the reasons you may be getting this error, and walk you through some of the steps you can take to try and resolve it as well.
+
+The error message you'll see in your console when trying to make a request to an Appwrite backend will look something like this: `Access blocked by CORS policy.`
+
+## **Understanding CORS**
+
+Before we start debugging this, let's talk about what CORS error is.
+
+Without diving deep into the topic, CORS (Cross-Origin Resource Sharing ) is a mechanism that allows a server to specify which origins can access it. By origins, I mean URLs. Basically, site A, our server sitting at `myapi.com` won't allow request coming from site B, our client sitting at `myfrontend.com`.
+
+This happens because our server has not added site B, `myfrontend.com` to its list of allowed origins therefore, any request coming from a URL that is not listed in our server's allowed origins will be rejected by our CORS policy. The solution in this case would be to simply add `myfrontend.com` to the list of allowed origins.
+
+![CORS error appwrite](/images/blog/cors-error/cors_example.png)
+
+CORS is crucial because it provides a secure way to make requests across different origins. Without CORS, any website would be able to make requests to our server, and this would lead to major problems.
+
+
+## Why you are getting a CORS error
+
+Now let's try to figure this all out in the context of Appwrite and why you may be getting this error. Currently, I have listed three main reasons. If more arise, I will update the article to include them.
+
+1. Origin not set in Console
+2. Origin is set incorrectly
+3. Bad ID on request
+
+### 1 - Origin not set in Console
```

Comment:
Move all you headings on level higher

## -> #

### -> ##

We render # as ##